tutordiscovery/templates/discovery/build/discovery/Dockerfile

```diff
@@ -16,7 +16,7 @@ RUN useradd --home-dir /openedx --create-home --shell /bin/bash --uid ${APP_USER
 USER ${APP_USER_ID}
 
 ###### Git-clone course-discovery repo
-ARG DISCOVERY_REPOSITORY=https://github.com/edx/course-discovery.git
+ARG DISCOVERY_REPOSITORY='{{ COURSE_DISCOVERY_REPOSITORY }}'
```

Comment:
Can you please document this new setting in the README? (all settings must be documented)

Reply:
Just saw this while searching another thing, I believe the plugin must have `COURSE_DISCOVERY_REPOSITORY` value by default in `plugins.py` file. In case,  user does not provide any value, so plugin will build the image from the `https://github.com/edx/course-discovery.git` url.

Reply:
@hinakhadim https://github.com/overhangio/tutor/pull/927

Reply:
Hina is right, the setting should be moved here.

Reply:
Thanks Hina for pointing this out

Reply:
Regis we don't have any `defaults.yml` file in discovery like tutor has where I can place this setting. 
Should I create one `defaults.yml` file for discovery, Will discovery pick this setting if I create in discovery.
As far as I know settings are picked from `tutor/tutor/templates/config/defaults.yml `

Reply:
The package has a Python entrypoint named "tutor.plugin.v1" in `setup.py` that points to the "tutordiscovery.plugin" module. Thus, `tutordiscovery/plugin.py` is imported at runtime to enable the plugin. Inside this module, you will see that the plugin uses the filters and actions from `tutor.hooks`. In particular, the values from `config["defaults"]` are added to the `tutor.hooks.Filters.CONFIG_DEFAULTS` filter. See here what this filter does: https://docs.tutor.overhang.io/reference/api/hooks/catalog.html#tutor.hooks.Filters.CONFIG_DEFAULTS

Thus, you should add new config defaults to `tutordiscovery.plugin` in `config["defaults"]`.